components/bl-granim-component/src/index.js

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useRef } from "react";
+
+import { s } from "./lib/granim.min";
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const DIRECTIONS = ["left-right", "diagonal", "top-bottom", "radial", "custom"];
+const BLENDING_MODE = [
+  "multiply", "screen", "normal", "overlay", "darken", "lighten", "lighter", "color-dodge", "color-burn",
+  "hard-light", "soft-light", "difference", "exclusion", "hue", "saturation", "color", "luminosity",
+];
+
+export default function GranimComponent({ component, elRef, eventHandlers, pods }) {
+  const {
+    style, classList, display, direction, customDirection, states, isPausedWhenNotInView, scrollDebounceThreshold,
+    stateTransitionSpeed, imageSource, imagePosX, imagePosY, stretchModeX, stretchModeY, blendingMode,
+  } = component;
+  const { onStart, onGradientChange, onEnd } = eventHandlers;
+
+  const granimContentPod = pods["granimContent"];
+
+  const canvasInstRef = useRef(null);
+  const canvasRef = useRef(null);
+
+  useEffect(() => {
+    if (direction === "custom" && !customDirection) {
+      return;
+    }
+
+    if (!states) {
+      return;
+    }
```

Comment:
```suggestion
    if (states) {
      <all code below>
    }
```

This for me looks better, but you could choose.